### PR TITLE
PGT 3.5

### PR DESCRIPTION
**Pull request by lldelisle**
- [x] The _version.py needs to be updated for pypi deploy.

### CODE REVIEW

**Comment by lldelisle** [comment]
# 3.5

## Enhancements:

- pyGenomeTracks goes much faster:

    - We now use bedtools to load only a portion of bed, gtf, bedgraph, narrowPeak, bedgraphMatrix, Epilogos, links. This speeds up the process dramatically especially for gtf files.

    - When you use a cool matrix, only the plotted region is loaded, this speeds up the process. If multiple regions on different chromosomes are provided in a BED this option is not used.

    - Unfortunately, it is not possible to speed the process with h5 matrices so you may want to convert them to cool with [hicConvertFormat](https://hicexplorer.readthedocs.io/en/latest/content/tools/hicConvertFormat.html>).

- pyGenomeTracks does not require sorted bed anymore.

- For coverage tracks (bedgraph and bigwig), there is a new parameter: ``grid`` which allows to display horizontal lines.

- For links, you can choose to use the middle of start1 and end1 and the middle of start2 and end2 thanks to ``use_middle`` instead of the extremities coordinates.

- For Hi-C matrices, mcool files are now supported.

- For vlines, the line_width can now be set.

## Minor enhancements:

- Bedgraph with NA values in the 4-th column do not raise error anymore.

- There is a progress bar for links.

- For Hi-C matrices, 

    - there is no error when the plot goes over the chromosome size provided by the matrix.

    - if the depth is smaller than binsize it makes an empty plot instead of raising errors.

- When the ``overlay_previous`` is set in the first track it is ignored and a warning is given instead of giving an error with a not meaningful error message.

- The color for negative values in coverage tracks (bigwig and bedgraph) is now part of the output of ``make_tracks_file``.

- Others output of ``make_tracks_file`` have been modified for better clarity.

- Others code related enhancements which are transparent to the users.

- The format of bed12 is more stringent (the blocks must span the chromStart and chromEnd).

## Bugfixes:

- For bed tracks when ``gene_rows`` or ``global_max_row`` was set the last row was very close to the bottom (sometimes even overlaying the next track). Now it is vertically centered and do not go over the track height.

- Solve some bugs how non conventional bed files are handle.

- Display more meaningful error messages.

- Only plot scale bar when in the region of plot.

- Do not show a warning when file_type is set for ``x_axis`` or ``spacer``.

**Comment by lldelisle** [comment]
Note: on linux the current version of libopenblas make segmentation faults (on my laptop and on travis at least).
For the bioconda recipe, I propose that we modify it to add:
```
libopenblas < 0.3.10
```
To avoid issues and we will relax it in the next release if it is fixed.

**Comment by lldelisle** [comment]
I would say, let's go...

File: README.md [hunk 1]
```diff
@@ -39,10 +39,15 @@ Installation
...
+The recommended way to install pyGenomeTracks is via conda
+
+```bash
+$ conda create -n pygenometracks -c bioconda -c conda-forge pygenometracks pyhon=3.7
+```
+To get a specific version, one can specify it. For example:
 
 ```bash
-$ conda install -c bioconda -c conda-forge pygenometracks
+$ conda create -n pygenometracks -c bioconda -c conda-forge pygenometracks=3.5 pyhon=3.7
```
Comment (joachimwolff) [inline README.md:50]: it is python

File: docs/content/installation.rst [hunk 2]
```diff
@@ -19,15 +19,20 @@ Requirements
...
-follow the directions for its installation. All of the requirements for pyGenomeTracks can be installed in Anaconda with:
+Command line installation using ``conda``
+-----------------------------------------
+
+We encourage users to use ``conda`` installation to install pygenometracks. All of the requirements for pyGenomeTracks can be installed via Anaconda with:
 
 .. code:: bash
 
-    $ conda install -c bioconda -c conda-forge pygenometracks
+    $ conda create -n pygenometracks -c bioconda -c conda-forge pygenometracks pyhon=3.7
```
Comment (joachimwolff) [inline docs/content/installation.rst:29]: python

File: docs/content/installation.rst [hunk 3]
```diff
@@ -19,15 +19,20 @@ Requirements
...
+
+To get a specific version, one can specify it. For example:
+
+.. code:: bash
+
+    $ conda create -n pygenometracks -c bioconda -c conda-forge pygenometracks=3.5 pyhon=3.7
```
Comment (joachimwolff) [inline docs/content/installation.rst:35]: python

File: docs/content/releases/3.3.rst [hunk 4]
```diff
@@ -2,25 +2,31 @@
 ===
 
 Release 3.3 adds:
+^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (joachimwolff) [inline docs/content/releases/3.3.rst:5]: Too long

File: docs/content/releases/3.3.rst [hunk 5]
```diff
@@ -2,25 +2,31 @@
...
    - a new gene type: ``tssarrow``
    - ``border_color`` and ``color_utr`` can be set to ``bed_rgb``.
    - ``all_labels_inside`` allow to put the name of the region on the left if it ends after the plotted region.
    - ``labels_in_margin`` allow to put the name of the region in the margin if it ends after the plotted region.
+
 - ``line_width`` can now be set in the narrowPeak tracks.
 - ``colormap`` can now be set in bedgraphMatrix tracks.
 
 bug fixes:
+^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (joachimwolff) [inline docs/content/releases/3.3.rst:29]: too long

File: docs/content/releases/3.4.rst [hunk 6]
```diff
@@ -2,22 +2,32 @@
...
 
 Improved documentation:
+^^^^^^^^^^^^^^^^^^^^^^^
 
 - fix url of test_data thanks to @sonnynguyen
 - Better documentation for pyGenomeTracks / makeTracksFile
 - Documentation of each track
 
 Enhancements:
+^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (joachimwolff) [inline docs/content/releases/3.4.rst:12]: too long

File: docs/content/releases/3.4.rst [hunk 7]
```diff
@@ -2,22 +2,32 @@
...
+
+    - operation on files: You can plot the result of an operation on one or two files without preprocessing. To use it, put the operation in ``operation`` with the variable ``file`` or both variables ``file`` and ``second_file``, for example ``operation = 0.89 * file`` or ``operation = max(file, second_file)`` or ``operation = file - second_file``. In the two later cases, you need to put the path of the second file in ``second_file``. However, this operation can be time consuming if you do it multiple times and you may prefer to convert your files in advance.
+
+    For more examples on these two new features, visit `the examples <https://pygenometracks.readthedocs.io/en/latest/content/examples.html#log-transform-and-operation-examples>`_.
+
 - Both ``type = line`` and ``type = lines`` are working in bedgraph_matrix
 
 Bugfix:
+^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (joachimwolff) [inline docs/content/releases/3.4.rst:31]: too long

File: docs/content/releases/3.5.rst [hunk 8]
```diff
@@ -0,0 +1,63 @@
+3.5
+===
+
+Enhancements:
+^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (joachimwolff) [inline docs/content/releases/3.5.rst:5]: too long

File: docs/content/releases/3.5.rst [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+- For coverage tracks (bedgraph and bigwig), there is a new parameter: ``grid`` which allows to display horizontal lines.
+
+- For links, you can choose to use the middle of start1 and end1 and the middle of start2 and end2 thanks to ``use_middle`` instead of the extremities coordinates.
+
+- For Hi-C matrices, mcool files are now supported.
+
+- For vlines, the line_width can now be set.
+
+Minor enhancements:
+^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (joachimwolff) [inline docs/content/releases/3.5.rst:26]: too long

File: docs/content/releases/3.5.rst [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+- Others output of ``make_tracks_file`` have been modified for better clarity.
+
+- Others code related enhancements which are transparent to the users.
+
+- The format of bed12 is more stringent (the blocks must span the chromStart and chromEnd).
+
+- Improve the documentation regarding the installation. We highly recommand to use conda.
+
+Bugfixes:
+^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (joachimwolff) [inline docs/content/releases/3.5.rst:51]: too long

File: docs/content/tracks/auto/bed_options_text.txt [hunk 11]
```diff
@@ -5,17 +5,18 @@ title =
...
 #overlay_previous = yes
 
 # If the bed file contains the exon
 # structure (bed 12) then this is plotted. Otherwise
 # a region **with direction** is plotted.
 # If the bed file contains a column for color (column 9), then this color can be used by
 # setting:
-# color = bed_rgb
-#if color is a valid colormap name (like RbBlGn), then the score is mapped
+#color = bed_rgb
```
Comment (joachimwolff) [inline docs/content/tracks/auto/bed_options_text.txt:18]: space or no space?

Comment (lldelisle) [reply docs/content/tracks/auto/bed_options_text.txt:18]: I put space for explanations and no space when it is something that can be used.

Comment (joachimwolff) [reply docs/content/tracks/auto/bed_options_text.txt:18]: :+1:

**Pull request merged**